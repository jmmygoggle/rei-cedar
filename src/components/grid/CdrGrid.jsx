import clsx from 'clsx';
import propValidator from '../../utils/propValidator';
import style from './styles/CdrGrid.scss';

export default {
  name: 'CdrGrid',
  props: {
    /**
     * Defines gutter size.
     * Possible values: {none, small, medium, large}.
     * Also accepts responsive values with `@breakpoint`: "none@md"
     */
    gutter: {
      type: String,
      validator: (value) => propValidator(
        value,
        ['none', 'small', 'medium', 'large'],
      ),
      default: 'medium@xs medium@sm large@md large@lg',
    },
    tag: {
      type: String,
      default: 'div',
    },
  },
  data() {
    return {
      style,
    };
  },
  computed: {
    gutterClass() {
      const classStr = [];
      if (this.gutter) {
        this.gutter.split(' ').forEach((val) => {
          classStr.push(this.style[`cdr-grid--gutter-${val}`]);
        });
      }
      return classStr.join(' ');
    },
  },
  render() {
    const Component = this.tag;
    return (
      <Component class={clsx(this.style['cdr-grid'], this.gutterClass)}>
        {this.$slots.default}
      </Component>
    );
  },
};
