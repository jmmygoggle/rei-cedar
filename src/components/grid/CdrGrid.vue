<template>
  <component
    :is="tag"
    :class="[$style[baseClass], $style[gutterClass]]"
  >
    <slot />
  </component>
</template>

<script lang="ts">
import { defineComponent, computed } from 'vue';
import propValidator from '../../utils/propValidator';

export default defineComponent({
  name: 'CdrGridTwo',
  props: {
    /**
     * Defines gutter size.
     * Possible values: {none, small, medium, large}.
     * Also accepts responsive values with `@breakpoint`: "none@md"
     */
    gutter: {
      type: String,
      validator: (value) => propValidator(
        value,
        ['none', 'small', 'medium', 'large'],
      ),
      default: 'medium@xs medium@sm large@md large@lg',
    },
    tag: {
      type: String,
      default: 'div',
    },
  },
  setup(props) {
    const baseClass = 'cdr-grid-two';
    // TODO: buildClass refactor
    const gutterClass = computed(() => props.gutter
      && props.gutter.split(' ').map((val) => `cdr-grid-two--gutter-${val}`).join(' '));

    return {

      baseClass,
      gutterClass,
    };
  },
});

</script>

<style lang="scss" module src="./styles/CdrGridTwo.scss">
</style>
