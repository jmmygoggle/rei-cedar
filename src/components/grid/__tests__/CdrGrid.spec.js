import { shallowMount } from '@vue/test-utils';
import CdrGrid from 'componentdir/grid/CdrGrid';

describe('CdrGrid', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(CdrGrid);
    expect(wrapper.element).toMatchSnapshot();
  });


  it('complex example matches snapshot', () => {
    const wrapper = shallowMount(CdrGrid, {
      propsData: {
        gutter: 'none@xs large@sm medium@md small@sm',
        tag: 'ul'
      },
      slots: {
        default: '<li>hey</li>'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
