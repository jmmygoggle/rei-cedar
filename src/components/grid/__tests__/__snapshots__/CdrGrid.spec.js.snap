// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CdrGrid complex example matches snapshot 1`] = `
<ul
  class="cdr-grid cdr-grid--gutter-none@xs cdr-grid--gutter-large@sm cdr-grid--gutter-medium@md cdr-grid--gutter-small@sm"
>
  <li>
    hey
  </li>
</ul>
`;

exports[`CdrGrid matches snapshot 1`] = `
<div
  class="cdr-grid cdr-grid--gutter-medium@xs cdr-grid--gutter-medium@sm cdr-grid--gutter-large@md cdr-grid--gutter-large@lg"
/>
`;
